src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -236,6 +236,29 @@ private MessagePayloadDataConverter CreateErrorDataConverter(IErrorSerializerSet
             return new MessagePayloadDataConverter(errorSerializerSettingsFactory.CreateJsonSerializerSettings(), isDefault);
         }
 
+        private IDurabilityProviderFactory GetDurabilityProviderFactory(ILogger logger, IEnumerable<IDurabilityProviderFactory> orchestrationServiceFactories)
+        {
+            bool storageTypeIsConfigured = this.Options.StorageProvider.TryGetValue("type", out object storageType);

```

Comment:
I would almost prefer if the `DurableTaskOptions` was directly passed into a parameter for this method. I realize that we have already set `this.Options` has already been set at this point, but it's possible during a refactor that this method gets called when this is null, resulting in a hard to diagnose NullReferenceException.